exonum-java-binding-core/src/main/java/com/exonum/binding/transaction/TransactionContext.java

```diff
@@ -0,0 +1,92 @@
+/*
+ * Copyright 2018 The Exonum Team
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.exonum.binding.transaction;
+
+import com.exonum.binding.common.crypto.PublicKey;
+import com.exonum.binding.common.hash.HashCode;
+import com.exonum.binding.storage.database.Fork;
+
+/**
+ * Transaction context class. Contains required information for the transaction execution.
+ */
+public interface TransactionContext {
+  /**
+   * Returns database view allowing R/W operations.
+   */
+  Fork getFork();
+
+  /**
+   * Returns SHA-256 hash of the transaction message.
+   */
+  HashCode getTransactionMessageHash();
+
+  /**
+   * Returns public key of the transaction author.
+   */
+  PublicKey getAuthorPk();
+
+  /**
+   * Returns the builder of the transaction context.
+   */
+  static Builder builder() {
+    return new Builder();
+  }
+
+  /**
+   * Transaction context builder.
+   */
+  final class Builder {
+    private Fork fork;
+    private HashCode hash;
+    private PublicKey authorPk;
+
+    /**
+     * Sets database fork for the context.
+     */
+    public Builder fork(Fork fork) {
+      this.fork = fork;
+      return this;
+    }
+
+    /**
+     * Sets message hash for the context.
+     */
+    public Builder hash(HashCode hash) {
```

Comment:
From the usages I've seen in tests, it seems that a more exact name would be better,
     because it is not clear what is a hash of a transaction execution context. `messageHash`?
     `txMessageHash`?